### PR TITLE
Fix comparison with nan

### PR DESCRIPTION
**Pull request by Gobot1234**
Also fixed some typos in the docstrings

### CODE REVIEW

**Review by cetanu — DISMISSED** [review]

File: src/betterproto/__init__.py [hunk 1]
```diff
@@ -455,20 +455,19 @@ def _dump_float(value: float) -> Union[float, str]:
...
-        Dumped valid, either a float or the strings
-        "Infinity" or "-Infinity"
+        Dumped value, either a float or the strings
     """
     if value == float("inf"):
         return INFINITY
     if value == -float("inf"):
         return NEG_INFINITY
-    if value == float("nan"):
+    if math.isnan(value):
```
Comment (abn) [inline src/betterproto/__init__.py:470]:
```suggestion
    if isinstance(value, float) and math.isnan(value):
```

I am not sure why the test case ended up passing a `None` value in the first place when this method does not expect it.

**Review by abn — CHANGES_REQUESTED** [review]

**Review by cetanu — APPROVED** [review]

File: src/betterproto/__init__.py [hunk 2]
```diff
@@ -455,20 +455,19 @@ def _dump_float(value: float) -> Union[float, str]:
...
-        Dumped valid, either a float or the strings
-        "Infinity" or "-Infinity"
+        Dumped value, either a float or the strings
     """
     if value == float("inf"):
         return INFINITY
     if value == -float("inf"):
         return NEG_INFINITY
-    if value == float("nan"):
+    if isinstance(value, float) and math.isnan(value):
```
Comment (abn) [inline src/betterproto/__init__.py:470]: @Gobot1234 should we skip the other checks as well if the instance is not a float? Seems pointless if it is not a float.

Comment (Gobot1234) [reply src/betterproto/__init__.py:470]: I think a guard against None in to/from_dict would be the best solution

**Review by abn — APPROVED** [review]

File: src/betterproto/__init__.py (hunk 2 above)
Comment (Gobot1234) [reply src/betterproto/__init__.py:470]: I'll do this in #367 just cause I want to get this merged

**Pull request merged**